src/Helpers/UpdateHelper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Acquia\Cli\Helpers;
+
+use Acquia\Cli\SelfUpdate\Strategy\GithubStrategy;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Humbug\SelfUpdate\Updater;
+use Kevinrob\GuzzleCache\CacheMiddleware;
+use Phar;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Webmozart\PathUtil\Path;
+
+class UpdateHelper {
+
+  /** @var string */
+  protected $pharPath;
+
+  /**
+   * @var \GuzzleHttp\Client
+   */
+  protected $client;
+
+  /**
+   * @param \GuzzleHttp\Client $client
+   */
+  public function setClient($client): void {
+    $this->client = $client;
+  }
+
+  /**
+   * @return \GuzzleHttp\Client
+   */
+  public function getClient(): Client {
+    if (!isset($this->client)) {
+      $stack = HandlerStack::create();
+      $stack->push(new CacheMiddleware(), 'cache');
```

Comment:
@anavarre Adding caching here. It should respect GitHub's cache headers. Will need to validate that it's actually limiting outgoing requests.

Reply:
Thanks, I'm watching this PR. When you think it's ready, let me know how I can test it, maybe?